More THM docs: materials, actions, scalaraux, vpps, some pps

GiudGiud opened the pull request:
credits to SL express for taking so long

A lot of the objects are kind of trivial so the doco page is quite laminar as well. It's still better than a stub
Please point out the ones you think could use more details or an "example input syntax" to add more meat

GiudGiud committed:
Document all auxscalar kernels from THM, refs #19737

GiudGiud committed:
Document THM actions that have a stub, refs #19737

GiudGiud committed:
Document the two matprop samplers in THM, refs #19737

GiudGiud committed:
Link to regular doco for AD DGkernel in THM
Clean stub for unused massfree constraint

GiudGiud committed:
Document most of the materials in THM, refs #19737
- numerous materials are duplicated between regular and AD
Documentation from AD points to the regular one to avoid duplication & its maintenance cost

GiudGiud committed:
Add class descriptions to some THM materials
refs #15968

GiudGiud committed:
Add user docs for some THM postprocessors, refs  #19737

GiudGiud committed:
Add missing class descriptions to some THM postprocessors
refs #15968

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1594821/) on 07fdb27 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/24725/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/24725/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 20e735bcdd25ae44efc67790e23091d2eed7ff00`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1594861/) on 2ee43fb wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24725/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1594902/) on 2ee43fb wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>20e735</th>
      <th colspan=4>#24725 2ee43f</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.41%</td>
      <td>85.40%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>88937</td>
      <td>88932</td>
      <td>-5</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15196</td>
      <td>15201</td>
      <td>+5</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24725/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24725/coverage/framework)


## Modules coverage

### Thermal hydraulics

<table>
  <thead>
    <tr>
      <th></th>
      <th>20e735</th>
      <th colspan=4>#24725 2ee43f</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>90.37%</td>
      <td>90.37%</td>
      <td>-0.00%</td>
      <td>94.83%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>12534</td>
      <td>12562</td>
      <td>+28</td>
      <td>55</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1335</td>
      <td>1338</td>
      <td>+3</td>
      <td>3</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24725/coverage_diff/thermal_hydraulics)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24725/coverage/thermal_hydraulics)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24725/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud commented:
Time stepper diff showing up here https://civet.inl.gov/job/1595089/

joshuahansel commented on modules/thermal_hydraulics/doc/content/source/postprocessors/MomentumFluxIntegral.md:
```diff
@@ -1,20 +1,14 @@
 
-## Example Input File Syntax
+!equation
+I_M = \int_{\partial \Omega} arhouA * velocity + \alpha * p * A d\partial \Omega
```
```suggestion
I_M = \int_{\partial \Omega} \alpha \rho u^2 A + \alpha p A d\partial \Omega
```

joshuahansel commented on modules/thermal_hydraulics/doc/content/source/postprocessors/MomentumFluxIntegral.md:
```diff
@@ -1,20 +1,14 @@
+I_M = \int_{\partial \Omega} arhouA * velocity + \alpha * p * A d\partial \Omega
 
-!! Describe and include an example of how to use the MomentumFluxIntegral object.
+with $arhouA$ the conserved phase momentum, $velocity$ the boundary 1D velocity,
```
```suggestion
with $\alpha \rho u A$ the conserved phase momentum, $u$ the boundary 1D velocity,
```

GiudGiud committed:
Address Josh's review

Oxford commas
No asterisk for products in math expressions
Remove random letters from me trying to type in the console but the text editor was still the active window

Co-authored-by: joshuahansel <joshua.hansel@inl.gov>

GiudGiud commented:
Accepted all, thanks for spotting them!

joshuahansel commented:
Thanks @GiudGiud/SL Express for doing all of these - I can't imagine it was too exciting...

GiudGiud commented:
so green so beautiful

joshuahansel reviewed (approved):
So approved
